zerver/data_import/mattermost.py

```diff
@@ -433,46 +477,67 @@ def process_batch(lst: List[Dict[str, Any]]) -> None:
 def write_message_data(num_teams: int,
                        team_name: str,
                        realm_id: int,
-                       post_data: List[Dict[str, Any]],
+                       post_data: Dict[str, List[Dict[str, Any]]],
                        zerver_recipient: List[ZerverFieldsT],
                        subscriber_map: Dict[int, Set[int]],
                        output_dir: str,
                        masking_content: bool,
                        stream_id_mapper: IdMapper,
+                       huddle_id_mapper: IdMapper,
                        user_id_mapper: IdMapper,
                        user_handler: UserHandler,
                        username_to_user: Dict[str, Dict[str, Any]],
                        zerver_realmemoji: List[Dict[str, Any]],
                        total_reactions: List[Dict[str, Any]]) -> None:
-
-    stream_id_to_recipient_id = {
-        d['type_id']: d['id']
-        for d in zerver_recipient
-        if d['type'] == Recipient.STREAM
-    }
-
-    def get_stream_recipient_id(raw_message: ZerverFieldsT) -> int:
-        receiver_id = raw_message['receiver_id']
-        stream_id = stream_id_mapper.get(receiver_id)
-        recipient_id = stream_id_to_recipient_id[stream_id]
+    recipient_id_to_recipient_type = {}
+    stream_id_to_recipient_id = {}
+    huddle_id_to_recipient_id = {}
+    user_id_to_recipient_id = {}
+
+    for d in zerver_recipient:
+        recipient_id_to_recipient_type[d['id']] = d['type']
+        if d['type'] == Recipient.STREAM:
+            stream_id_to_recipient_id[d['type_id']] = d['id']
+        elif d['type'] == Recipient.HUDDLE:
+            huddle_id_to_recipient_id[d['type_id']] = d['id']
+        if d['type'] == Recipient.PERSONAL:
+            user_id_to_recipient_id[d['type_id']] = d['id']
+
+    def get_recipient_id_from_receiver_name(receiver_name: str, recipient_type: int) -> int:
+        if recipient_type == Recipient.STREAM:
+            receiver_id = stream_id_mapper.get(receiver_name)
+            recipient_id = stream_id_to_recipient_id[receiver_id]
+        elif recipient_type == Recipient.HUDDLE:
+            receiver_id = huddle_id_mapper.get(receiver_name)
+            recipient_id = huddle_id_to_recipient_id[receiver_id]
+        elif recipient_type == Recipient.PERSONAL:
+            receiver_id = user_id_mapper.get(receiver_name)
+            recipient_id = user_id_to_recipient_id[receiver_id]
```

Comment:
Again, an `else raise AssertionError` here would be good.

Reply:
Done.